src/p11_child/p11_child_openssl.c

```diff
@@ -42,8 +43,325 @@ struct p11_ctx {
     X509_STORE *x509_store;
     const char *ca_db;
     bool wait_for_card;
+    struct cert_verify_opts *cert_verify_opts;
 };
 
+static OCSP_RESPONSE *query_responder(BIO *cbio, const char *host,
+                                      const char *path,
+                                      OCSP_REQUEST *req, int req_timeout)
+{
+    int fd;
+    int rv;
+    OCSP_REQ_CTX *ctx = NULL;
+    OCSP_RESPONSE *rsp = NULL;
+    fd_set confds;
+    struct timeval tv;
+
+    if (req_timeout != -1) {
+        BIO_set_nbio(cbio, 1);
+    }
+
+    rv = BIO_do_connect(cbio);
+
+    if ((rv <= 0) && ((req_timeout == -1) || !BIO_should_retry(cbio))) {
```

Comment:
Why do we call should_retry if we never retry? Is it for future extension?

Reply:
I took inspiration from the related code from apps/ocsp.c form the OpenSSL source code. If I understood the man page correctly there might be cases when SSL is used where retry is returned but can be ignored.